ykrt/src/mt.rs

```diff
@@ -629,6 +631,21 @@ impl MT {
                             }
                         }
                         HotLocationKind::Compiling => TransitionControlPoint::NoAction,
+                        HotLocationKind::Counting(c) => {
+                            if is_tracing {
+                                // This thread is tracing something, so bail out as quickly as possible
+                                TransitionControlPoint::NoAction
+                            } else {
+                                if c < self.hot_threshold() {
```

Comment:
So we retry compiling a trace as many times as the hot threshold that triggered the first compilation? I wonder if there should be a separate threshold at some point?

Reply:
> So we retry compiling a trace as many times as the hot threshold that triggered the first compilation?

No, we retry `TraceFailed` times. Each of those is spaced out by `HotThreshold` times.

Reply:
> we retry TraceFailed times

At the risk of sounding like dummy: I'm not sure what you mean.

`TraceFailed` is a type:
```
pub(crate) enum TraceFailed {
    KeepTrying,
    DontTrace,
}
```

Reply:
Oops, I meant `TraceCompilationErrorThreshold.

Reply:
OK, I see it. That logic was already in place, but I think we are saying that due to a bug (getting stuck in `Compiling`) it wasn't used properly.

Reply:
It was used in one place but not in another (there are lots of ways that things can go wrong :)). This PR fixes the latter case.